sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -0,0 +1,647 @@
+myparams::joint_matrix_a sub_a(sg);
+myparams::joint_matrix_b sub_b(sg);
+myparams::joint_matrix_c sub_c(sg);
+
+```
+
+=== General Query Example:
+```c++
+constexpr int M = 1500; // with msize = 8 and msize = 4,
+          // M can be broken up to 125 sequence of 8-sized ops and remaining 500 using 125 sequence of 4-sized ops
+tpu_params<tpu::dpas> params;
+constexpr int msize = break_dimension(params, M);
+constexpr int msize_remainder = break_dimension_remainder(params, M);
+constexpr int nsize = params.combinations[0].nsize;
+constexpr int ksize = params.combinations[0].ksize;
+// device code:
+joint_matrix<int8_t, use::a, msize, ksize> sub_a(sg);
+joint_matrix<int8_t, use::b, ksize, nsize> sub_b(sg);
+joint_matrix<int, use::accumulator, msize, nsize> sub_c(sg);
+//Remainder handling
+```
+
+## Future-looking API
+
+### Memory scope
+The current experimental API uses `joint_` semantics to define the memory scope of the matrix. The long term solution is to use the proposed link:../supported/sycl_ext_oneapi_local_memory.asciidoc[`group_local_memory` extension] to allocate the matrix in local memory associated with a SYCL group as shown in the example below.
+
+
+```c++
+multi_ptr<matrix<T>, address_space::local_space> tA_ptr = group_local_memory<matrix<sub_group, int8_t, tM, tN, use::a>>(sg);
+```
+We did not utilize this extension for this matrix API version because sub-group local memory is not yet well defined in {dpcpp}. Moreover, the representation of this notion in LLVM IR and SPIR-V is not clear yet. 
+
+### WI data to joint matrix mapping coordinates information for piece-wise operations
+The indexing provided inside the `wi_data` class acesses only the portion of the current WI. It is not possible the location or coordinates of this portion in the original matrix.  This coordinates mapping  is implementation defined and change from one backend to the other.   For general piece-wise operations like sum of rows of a matrix, the WI data to joint matrix mapping coordinates information is needed to reason about the matrix view.
+With joint matrix, we want to write, as much as possible, one code to run on different backends. So if backend X states that a WI owns one exact row of the matrix for instance. Writing the following code will work only on that backend for that version of hardware. The hardware and implementations change, for instance, the same WI can own half of the row because SG size increased or hardware units increased. 
+
+```c++
+auto data = C.get_wi_data();
+for (int i = 0; i < length; ++i) {
+  sum_of_local_rows[row] += data[i];
+}
+```
+
+
+
+We want to keep backward compatibility in the joint matrix code when implementations or hardware change. To that end, instead of hard-code this mapping, we write  general backend and target-agnostic, especially in the JIT compilation mode of SYCL. This is possible by querying this mapping so code does not have to change from one version to the other.
```

Comment:
```suggestion
We want to keep backward compatibility in the joint matrix code when implementations or hardware change. To that end, instead of hard-coding this mapping, we use general backend and target-agnostic functionality, especially in the JIT compilation mode of SYCL. This is possible by querying this mapping so code does not have to change from one version to the other.
```

I find the last sentence confusing "This is possible by querying this mapping so code does not have to change from one version to the other"

Reply:
sounds good